tests/compiler/test_index.py

```diff
@@ -141,9 +149,19 @@ def test_issue_212_bad_index_defintion(snippetcompiler):
 def test_index_on_subtype(snippetcompiler):
     snippetcompiler.setup_for_snippet(
         """
+        entity A:
+            string path
+        end
+
+        A.host [1] -- std::Host
+
+        index A(host, path)
+
+        implement A using std::none
+
         host = std::Host(name="a",os=std::linux)
-        a=std::DefaultDirectory(host=host,path="/etc")
-        b=std::DefaultDirectory(host=host,path="/etc")
+        a=A(host=host,path="/etc")
+        b=A(host=host,path="/etc")
```

Comment:
This is removes the entire purpose of this test: to test if indexes on supertypes also apply to subtypes. There is not subtyping here.